source/yats/include/yats/Util.h

```diff
@@ -69,4 +69,61 @@ struct TaskHelper
 template <typename ReturnType, typename TaskType, typename... ParameterTypes>
 static constexpr TaskHelper<ReturnType, ParameterTypes...> MakeHelper(ReturnType(TaskType::*)(ParameterTypes...));
 
+template<uint64_t Id, typename T>
+struct get_element_by_id
+{
+	template<typename... Args>
+	struct Parser;
+
+	template<typename First, typename... Args>
+	struct Parser<First, Args...>
+	{
+		template<uint64_t LocalId = Id, typename F = First>
+		static constexpr decltype(std::enable_if_t<F::ID != LocalId, Parser<Args...>>::parse_type()) parse_type();
+
+		template<uint64_t LocalId = Id, typename F = First>
+		static constexpr typename std::enable_if_t<F::ID == LocalId, F>::value_type parse_type();
+	};
+
+	template<typename First>
+	struct Parser<First>
+	{
+		template<uint64_t LocalId = Id, typename F = First>
+		static constexpr typename std::enable_if_t<F::ID == LocalId, F>::value_type parse_type();
+	};
+
+	template<typename... Args>
+	static constexpr Parser<Args...> Help(std::tuple<Args...>);
+
+	template<size_t Index = 0, typename Tuple = T>
+	static constexpr std::enable_if_t<Index < std::tuple_size<Tuple>::value, size_t> outer_parse_index()
+	{
+		return inner_parse_index<Index>();
+	}
+
+	template<size_t Index = 0, typename Tuple = T>
+	static constexpr std::enable_if_t<Index == std::tuple_size<Tuple>::value, size_t> outer_parse_index()
+	{
+		return 0;
+	}
+
+	template<size_t Index, uint64_t LocalId = Id, typename Tuple = T>
+	static constexpr std::enable_if_t<std::tuple_element<Index, Tuple>::type::ID == LocalId, size_t> inner_parse_index()
+	{
+		return Index + outer_parse_index<Index + 1>();
+	}
+
+	template<size_t Index = 0, uint64_t LocalId = Id, typename Tuple = T>
+	static constexpr std::enable_if_t<std::tuple_element<Index, Tuple>::type::ID != LocalId, size_t> inner_parse_index()
+	{
+		return outer_parse_index<Index + 1>();
+	}
+
+	using type = decltype(decltype(Help(std::declval<T>()))::parse_type());
+	static constexpr uint64_t index = outer_parse_index();
+};
+
+template<uint64_t Id, typename T>
+using get_element_by_id_t = typename get_element_by_id<Id, T>::type;
```

Comment:
v

Reply:
If we follow the style of the standard library it should be _t since it is a type.

Reply:
but std::tuple_element_v is also a type and uses v